src/core/annotation.js

```diff
@@ -171,6 +171,7 @@ var Annotation = (function AnnotationClosure() {
     this.data.color = this.color;
     this.data.borderStyle = this.borderStyle;
     this.data.hasAppearance = !!this.appearance;
+    this.data.hasHtml = true;
   }
```

Comment:
It might be slightly more efficient (in terms of optimizations in the JS engine), if the shape of this object didn't change so much. Hence you might want to consider doing e.g.
```javascript
this.data = {
  id: ...
  subtype: ...
  ...
};
```
Also, adding `annotationType: null` in the above probably wouldn't hurt either.